benchmarks/distributed/rpc/ps/metrics/MetricsLogger.py

```diff
@@ -0,0 +1,92 @@
+from .CPUMetric import CPUMetric
+from .CUDAMetric import CUDAMetric
+
+
+class MetricsLogger:
+
+    CUDA = "cuda"
+    CPU = "cpu"
+
+    def __init__(self, rank, metric_class, overwrite_metrics=False):
+        self.rank = rank
+        self.metric_class = metric_class.lower()
+        self.overwrite_metrics = overwrite_metrics
+        self.metrics = {}
+
+    def add_metric(self, metric_type, key, metric_name):
+        if metric_type in self.metrics and key in self.metrics[metric_type] and not self.overwrite_metrics:
+            raise Exception("metric_type={} with key={} already exists".format(metric_type, key))
+
+        if metric_type not in self.metrics:
+            self.metrics[metric_type] = {}
+
+        if self.metric_class == self.CPU:
+            self.metrics[metric_type][key] = CPUMetric(metric_name)
+        elif self.metric_class == self.CUDA:
+            self.metrics[metric_type][key] = CUDAMetric(metric_name)
```

Comment:
If this the intention, I guess you can allow the user to directly pass in a metric class (not string) to the ctor. I.e.:

```python
def __init__(self, rank, metric_class):
    self.metric_class = metric_class

def add_metric(...):
    self.metrics[type][key] = self.metric_class(name)
```

And users can call the ctor with 

```python
logger = MetricsLogger(0, CUDAMetric)
```

Reply:
I think just allowing the logger to log both is a better solution. https://github.com/pytorch/pytorch/pull/57454#discussion_r626144963